stdlib/public/core/ArrayShared.swift

```diff
@@ -27,6 +27,7 @@ public struct _DependenceToken {
 /// - Precondition: `storage` is `_ContiguousArrayStorage`.
 @inlinable // FIXME(inline-always)
 @inline(__always)
+@_semantics("array.allocate_uninitialized")
 public // COMPILER_INTRINSIC
```

Comment:
I think this being a compiler intrinsic and also something declared in `KnownDecls.def`, we should not have a semantics attribute here but instead just compare to `astContext.getAllocateUninitializedArray()`. 

@eeckstein It would be very helpful if you can share your thoughts on this issue as well.